sky/templates/websocket_proxy.py

```diff
@@ -59,6 +60,11 @@ async def websocket_to_stdout(websocket):
 
 if __name__ == '__main__':
     server_url = sys.argv[1].strip('/')
-    websocket_url = (f'ws://{server_url}/kubernetes-pod-ssh-proxy'
-                     f'?cluster_name={sys.argv[2]}')
+    server_proto, server_fqdn = server_url.split('://')
+    websocket_proto = 'ws'
+    if server_proto == 'https':
+        websocket_proto = 'wss'
+    websocket_url = (
+        f'{websocket_proto}://{server_fqdn}/kubernetes-pod-ssh-proxy'
+        f'?cluster_name={sys.argv[2]}')
```

Comment:
```suggestion
    if '://' in server_url:
        server_proto, server_fqdn = server_url.split('://')
        websocket_proto = 'ws'
        if server_proto == 'https':
            websocket_proto = 'wss'
        server_url = f'{websocket_proto}://{server_fqdn}'
    else:
        # Keep backward compatibility for legacy server URLs without protocol
        # TODO(aylei): Remove this after 0.10.0
        server_url = f'ws://{server_url}'
    websocket_url = (
        f'{server_url}/kubernetes-pod-ssh-proxy'
        f'?cluster_name={sys.argv[2]}')
```

Maybe we should keep the handling of the legacy URL in case the SkyPilot upgraded but the ssh config is not yet refreshed. Wdyt?

Reply:
yeah that makes sense, good idea!